U8HF15/AI_Practice/Functions/LossFunctions/MAE.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AI_Practice.Functions.LossFunctions
+{
+    public static partial class LossFunctions
+    {
+        public static float[] MAE(float[] Preds, float[] Reals)
+        {
+            float[] res = new float[Preds.Length];
+            for (int k = 0; k < res.Length; k++)
+            {
+                res[k] = Preds[k] - Reals[k];
+            }
+            return [res.Sum(x => float.Abs(x)) / res.Length];
+             
+        }
+
+        public static float[] D_MAE(float[] Preds, float[] Reals)
```

Comment:
AI taskokhoz vannak egyébként beépített hardver gyorsított függvények: https://learn.microsoft.com/en-us/dotnet/api/system.numerics.tensors.tensorprimitives?view=net-9.0-pp A System.Numerics.Tensors telepítésével